components/payment.js

```diff
@@ -124,19 +124,19 @@ const useWebLnPayment = () => {
           .catch(reject)
       })
```

Comment:
Heads up: `waitUntilPaid`'s interval leaks. It polls even after our attempt to pay it fails and will keep polling until it expires.

I'm messing around with how to fix it, but it's tricky.

Reply:
I ended up using an `AbortController` in [a41c710](https://github.com/stackernews/stacker.news/pull/1195/commits/a41c710751c55647aadcc0da4dafe73c4c8a0d4a)

Reply:
> It polls even after our attempt to pay it fails and will keep polling until it expires.

Mhh, this means the invoice wasn't properly cancelled. Aren't we always cancelling it on failure? Any payment error or closing the modal should automatically cancel the invoice.

But I see the value in calling `stopWaiting` in the `finally` block to make sure. I just think our code should also cancel any invoices that we know are not going to get paid.